src/Agent.Worker/Worker.cs

```diff
@@ -17,6 +17,56 @@ public sealed class Worker : AgentService, IWorker
     {
         private readonly TimeSpan WorkerStartTimeout = TimeSpan.FromSeconds(30);
 
+        private void InitializeSecrets(JobRequestMessage message)
+        {
+            var secretMasker = HostContext.GetService<ISecretMasker>();
+            
+            // Add mask hints
+            if (message?.Environment?.MaskHints != null)
+            {
+                IDictionary<string, string> variables = message?.Environment?.Variables;
+
+                foreach (MaskHint maskHint in message.Environment.MaskHints)
+                {
+                    if (maskHint.Type == MaskType.Regex)
+                    {
+                        secretMasker.Add(new RegexMask(maskHint.Value));
+                    }
+                    else if (maskHint.Type == MaskType.Variable && variables != null)
+                    {
+                        string value;
+                        if (variables.TryGetValue(maskHint.Value, out value))
+                        {
+                            if (!string.IsNullOrEmpty(value))
+                            {
+                                secretMasker.Add(new VariableValueMask(maskHint.Value, value));
+                            }
+                        }
+                    }
+                }
+            }
+
+            // Add masks for service endpoints
+            if (message?.Environment?.Endpoints != null)
+            {
+                foreach (ServiceEndpoint endpoint in message.Environment.Endpoints)
+                {
+                    if (endpoint.Authorization != null &&
```

Comment:
check if any Endpoint.Data need to be mask as well. I am not sure, i will double check as well.